Userland/Libraries/LibHTTP/HttpRequest.cpp

```diff
@@ -182,9 +182,21 @@ Optional<HttpRequest> HttpRequest::from_raw_request(ReadonlyBytes raw_request)
     else
         return {};
 
-    request.m_resource = URL::percent_decode(resource);
     request.m_headers = move(headers);
 
+    String requested_url = URL::percent_decode(resource);
+    Vector<String> url_parts = requested_url.split_limit('?', 2, true);
+
+    request.m_url.set_cannot_be_a_base_url(true);
+    if (url_parts.size() == 2) {
```

Comment:
isn't this always true with `limit = 2` and `keep_empty = true`? :thinking:

Reply:
Not really, for example:
* for `/page.html?foo=bar` the size would be 2
* for `/page.html?` the size would be 2 (due to the `keep_empty=true`)
* for `/page.html` the size would be 1